src/server/engine_factory.rs

```diff
@@ -34,12 +32,12 @@ struct FactoryInner {
     root_db: Mutex<Option<RocksEngine>>,
 }
 
-pub struct KvEngineFactoryBuilder<ER: RaftEngine> {
+pub struct KvEngineFactoryBuilder {
     inner: FactoryInner,
-    router: Option<RaftRouter<RocksEngine, ER>>,
+    compact_event_sender: Option<Arc<dyn CompactedEventSender + Send + Sync>>,
```

Comment:
Why `Arc`?

Reply:
Well because this CompactedEventSender is supposed to be thread safe, so no need to use Box instead? 
It is much easier to handle clone() for Arc than for Box.

Reply:
Then why there is a `clone` method in `CompactedEventSender`?

Reply:
So in builder, we can use Arc's clone as it's more lightweight. But for TabletFactory, we use CompactedEventSender's clone so that each tablet can have its own Mutex for better performance.